testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/util/saml/CreateArtifactMessageStepBuilder.java

```diff
@@ -40,13 +50,60 @@ public CreateArtifactMessageStepBuilder(URI authServerSamlUrl, String issuer, Sa
 
     @Override
     public HttpUriRequest perform(CloseableHttpClient client, URI currentURI, CloseableHttpResponse currentResponse, HttpClientContext context) throws Exception {
-        lastArtifact = SamlProtocolUtils.buildArtifact(issuer);
+        lastArtifact = createArtifact(issuer);
         if (SamlClient.Binding.POST == requestBinding) {
             return sendArtifactMessagePost();
         }
         return sendArtifactMessageRedirect();
     }
 
+    /**
+     * Creates an artifact. Format is:
+     *
+     * SAML_artifact := B64(TypeCode EndpointIndex RemainingArtifact)
+     *
+     * TypeCode := 0x0004
+     * EndpointIndex := Byte1Byte2
+     * RemainingArtifact := SourceID MessageHandle
+     *
+     * SourceID := 20-byte_sequence, used by the artifact receiver to determine artifact issuer
+     * MessageHandle := 20-byte_sequence
+     *
+     * @param entityId the entity id to encode in the sourceId
+     * @return an artifact
+     * @throws ProcessingException
+     * @throws IOException
+     */
+    private String createArtifact(String entityId) throws ArtifactResolverProcessingException {
+        /** SAML 2 artifact type code (0x0004). */
+        final byte[] TYPE_CODE = { 0, 4 };
```

Comment:
May be this type code should be defined in a single place?

Reply:
Done